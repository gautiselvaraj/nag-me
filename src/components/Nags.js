import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { StaggeredMotion, spring } from 'react-motion';
import Nag from './Nag';
import Button from './Button';
import H3 from './H3';
import NagSuggest from '../containers/NagSuggest';
import SearchSort from '../containers/SearchSort';

const NoNag = styled.div`
  margin-bottom: 20px;
  margin-top: 20px;
  text-align: center;

  h3 {
    margin-bottom: 1rem;
  }
`;

const NagsUl = styled.ul`
  list-style-type: none;
  margin: 0;
  overflow: hidden;
  padding: 0.5rem;
  will-change: transform;
`;

const NagsLi = styled.li`
  cursor: pointer;
  margin-bottom: 0.5rem;
  position: relative;

  &:last-child {
    margin-bottom: 0;
  }
`;

const nagsSpring = { stiffness: 150, damping: 16 };

export default class Nags extends Component {
  static propTypes = {
    nags: PropTypes.arrayOf(PropTypes.object),
    nagNew: PropTypes.func.isRequired,
    nagEdit: PropTypes.func.isRequired,
    nagPause: PropTypes.func.isRequired,
    nagResume: PropTypes.func.isRequired,
    nagDelete: PropTypes.func.isRequired
  };

  state = { initiated: false };

  getInterpolatedStyles = prevInterpolatedStyles => {
    const endValue = prevInterpolatedStyles.map((_, i) => {
      return i === 0
        ? { t: spring(0, nagsSpring) }
        : { t: spring(prevInterpolatedStyles[i - 1].t, nagsSpring) };
    });
    return endValue;
  };

  componentDidMount() {
    this.timeoutTimer = setTimeout(
      () => this.setState({ initiated: true }),
      225
    );
  }

  componentWillUnmount() {
    clearTimeout(this.timeoutTimer);
  }

  render() {
    const {
      nags,
      nagNew,
      nagEdit,
      nagDelete,
      nagResume,
      nagPause,
      nagStatusUpdate
    } = this.props;
    const { initiated } = this.state;

    if (!initiated) {
      return null;
    }

    if (!nags.length) {
      return (
        <NoNag>
          <H3>Nothing to nag about</H3>
          <Button onClick={nagNew}>Create your first Nag</Button>
          <NagSuggest />
        </NoNag>
      );
    }

    return (
      <div>
        <SearchSort />
        <StaggeredMotion
          defaultStyles={Array.from(Array(nags.length)).map(i => ({ t: 100 }))}
          styles={this.getInterpolatedStyles}
        >
          {interpolatingStyles => (
            <NagsUl>
              {interpolatingStyles.map(
                ({ t }, i) =>
                  !!nags[i] && (
                    <NagsLi
                      style={{
                        WebkitTransform: `translateY(${t}px)`,
                        transform: `translateY(${t}px)`
                      }}
                      key={nags[i].id}
                    >
                      <Nag
                        nag={nags[i]}
                        nagEdit={nagEdit}
                        nagPause={nagPause}
                        nagResume={nagResume}
                        nagDelete={nagDelete}
                        nagStatusUpdate={nagStatusUpdate}
                      />
                    </NagsLi>
                  )
              )}
            </NagsUl>
          )}
        </StaggeredMotion>
      </div>
    );
  }
}
