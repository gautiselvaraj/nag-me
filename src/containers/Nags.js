import {connect} from 'react-redux';
import Nags from '../components/Nags';
import {nagNew, nagEdit, nagPause, nagResume, nagDelete} from '../actions/NagActions';

const mapStateToProps = state => ({
  nags: state.getIn(['nag', 'visibleList']).toJS()
});

const mapDispatchToProps = dispatch => ({
  nagNew: () => dispatch(nagNew()),
  nagEdit: nagId => dispatch(nagEdit(nagId)),
  nagPause: nagId => dispatch(nagPause(nagId)),
  nagResume: nagId => dispatch(nagResume(nagId)),
  nagDelete: nagId => dispatch(nagDelete(nagId))
});

export default connect(mapStateToProps, mapDispatchToProps)(Nags);
