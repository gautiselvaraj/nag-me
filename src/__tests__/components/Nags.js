import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { fromJS } from 'immutable';
import Nags from '../../components/Nags';
import 'jest-styled-components';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureMockStore([]);

describe('<Nags />', () => {
  let nagNew,
    nagEdit,
    nagPause,
    nagResume,
    nagDelete,
    nagStatusUpdate,
    originalSetTimeout,
    dateNow;

  const nagList = [
    {
      id: 1,
      title: 'Test Nag 1',
      firstNag: 1234567890,
      nextNag: 1234567890,
      repeats: '1 min',
      status: 'LIVE',
      createdAt: 1234567890
    },
    {
      id: 2,
      title: 'Test Nag 2',
      firstNag: 1234567890,
      nextNag: 1234567890,
      repeats: '10 min',
      status: 'PAUSED',
      createdAt: 1234567890
    }
  ];

  beforeEach(() => {
    nagNew = jest.fn();
    nagEdit = jest.fn();
    nagPause = jest.fn();
    nagResume = jest.fn();
    nagDelete = jest.fn();
    nagStatusUpdate = jest.fn();
  });

  beforeAll(() => {
    dateNow = window.Date.now;
    window.Date.now = jest.fn(() => 1517227842390);
    originalSetTimeout = window.setTimeout;
    window.setTimeout = (callback, time) => callback();
  });

  afterAll(() => {
    window.Date.now = dateNow;
    window.setTimeout = originalSetTimeout;
  });

  describe('empty nag list', () => {
    it('should match snapshot', () => {
      const tree = renderer
        .create(
          <Provider store={mockStore()}>
            <Nags
              nags={[]}
              nagNew={nagNew}
              nagEdit={nagEdit}
              nagPause={nagPause}
              nagResume={nagResume}
              nagDelete={nagDelete}
              nagStatusUpdate={nagStatusUpdate}
            />
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should have correct elements and children', () => {
      const nags = mount(
        <Provider store={mockStore()}>
          <Nags
            nags={[]}
            nagNew={nagNew}
            nagEdit={nagEdit}
            nagPause={nagPause}
            nagResume={nagResume}
            nagDelete={nagDelete}
            nagStatusUpdate={nagStatusUpdate}
          />
        </Provider>
      );

      expect(nags.find('h3').length).toBe(1);
      expect(nags.find('h3').text()).toBe('Nothing to nag about');
      expect(nags.find('button').length).toBe(6);
      nags
        .find('button')
        .first()
        .simulate('click');
      expect(nagNew.mock.calls.length).toBe(1);
    });
  });

  describe('populated nag list', () => {
    it('should match snapshot', () => {
      const tree = renderer
        .create(
          <Provider store={mockStore(fromJS({ query: null, sortBy: null }))}>
            <Nags
              nags={nagList}
              nagNew={nagNew}
              nagEdit={nagEdit}
              nagPause={nagPause}
              nagResume={nagResume}
              nagDelete={nagDelete}
              nagStatusUpdate={nagStatusUpdate}
            />
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should have correct elements and children', () => {
      const nags = mount(
        <Provider store={mockStore(fromJS({ query: null, sortBy: null }))}>
          <Nags
            nags={nagList}
            nagNew={nagNew}
            nagEdit={nagEdit}
            nagPause={nagPause}
            nagResume={nagResume}
            nagDelete={nagDelete}
            nagStatusUpdate={nagStatusUpdate}
          />
        </Provider>
      );

      expect(nags.find('ul').length).toBe(1);
      expect(nags.find('li').length).toBe(2);
    });
  });
});
