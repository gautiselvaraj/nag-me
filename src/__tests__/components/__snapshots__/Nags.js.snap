// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nags /> empty nag list should match snapshot 1`] = `
.c0 {
  margin-bottom: 20px;
  margin-top: 20px;
  text-align: center;
}

.c0 h3 {
  margin-bottom: 1rem;
}

.c1 {
  font-size: 1.2rem;
  margin-top: 0.35rem;
  margin-bottom: 0.35rem;
}

.c2 {
  border: 1px solid undefined;
  cursor: pointer;
  display: inline-block;
  padding: 0 .75rem;
  box-sizing: border-box;
  font-size: 0.85rem;
  line-height: 2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 0.5s;
  transition-duration: 0.5s;
  -webkit-transition-property: background-color;
  transition-property: background-color;
}

.c10 {
  border: 1px solid undefined;
  cursor: pointer;
  display: inline-block;
  padding: 0;
  box-sizing: border-box;
  font-size: 0.85rem;
  line-height: 2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 0.5s;
  transition-duration: 0.5s;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  border-radius: 50%;
  height: 2rem;
  text-align: center;
  width: 2rem;
}

.c10 i {
  position: relative;
  top: 2px;
}

.c3 {
  border-top: 1px solid;
  margin-top: 20px;
  padding-top: 10px;
  text-align: center;
}

.c4 {
  font-size: 1rem;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.c5 {
  list-style-type: none;
  margin: 0;
  overflow: hidden;
  padding: 0.5rem 0.5rem 1rem;
}

.c6 {
  border-radius: 4px;
  box-shadow: 0 0 2px rgba(0,0,0,0.25);
  margin-bottom: 0.5rem;
  padding: 10px;
  padding-right: 50px;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: left;
}

.c6:last-child {
  margin-bottom: 0;
}

.c7 {
  font-size: 0.95rem;
  margin-top: 0.15rem;
  margin-bottom: 0.15rem;
}

.c8 {
  font-size: 0.8rem;
  margin: 5px 0 0;
}

.c9 {
  bottom: 0;
  position: absolute;
  right: 10px;
  top: 10px;
}

.c11 {
  color: undefined;
  font-family: icons;
  font-size: 1rem;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  line-height: 1;
  speak: none;
  text-transform: none;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c11:before {
  content: "\\ea09";
}

<div
  className="c0"
>
  <h3
    className="c1"
  >
    Nothing to nag about
  </h3>
  <button
    className="c2"
    onClick={[Function]}
  >
    Create your first Nag
  </button>
  <div>
    <div
      className="c3"
    >
      <h5
        className="c4"
      >
        Or try the nags below
      </h5>
    </div>
    <ul
      className="c5"
    >
      <li
        className="c6"
        style={
          Object {
            "WebkitTransform": "translateY(100px)",
            "transform": "translateY(100px)",
          }
        }
      >
        <h6
          className="c7"
        >
          Drink water
        </h6>
        <p
          className="c8"
        >
          repeats every 
          20 min
        </p>
        <div
          className="c9"
        >
          <button
            className="c10"
            onClick={[Function]}
            title="Start this nag"
          >
            <i
              className="c11"
            />
          </button>
        </div>
      </li>
      <li
        className="c6"
        style={
          Object {
            "WebkitTransform": "translateY(100px)",
            "transform": "translateY(100px)",
          }
        }
      >
        <h6
          className="c7"
        >
          Relax your eyes
        </h6>
        <p
          className="c8"
        >
          repeats every 
          30 min
        </p>
        <div
          className="c9"
        >
          <button
            className="c10"
            onClick={[Function]}
            title="Start this nag"
          >
            <i
              className="c11"
            />
          </button>
        </div>
      </li>
      <li
        className="c6"
        style={
          Object {
            "WebkitTransform": "translateY(100px)",
            "transform": "translateY(100px)",
          }
        }
      >
        <h6
          className="c7"
        >
          Take a deep breath
        </h6>
        <p
          className="c8"
        >
          repeats every 
          1 hour
        </p>
        <div
          className="c9"
        >
          <button
            className="c10"
            onClick={[Function]}
            title="Start this nag"
          >
            <i
              className="c11"
            />
          </button>
        </div>
      </li>
      <li
        className="c6"
        style={
          Object {
            "WebkitTransform": "translateY(100px)",
            "transform": "translateY(100px)",
          }
        }
      >
        <h6
          className="c7"
        >
          Remember to smile
        </h6>
        <p
          className="c8"
        >
          repeats every 
          4 hour
        </p>
        <div
          className="c9"
        >
          <button
            className="c10"
            onClick={[Function]}
            title="Start this nag"
          >
            <i
              className="c11"
            />
          </button>
        </div>
      </li>
      <li
        className="c6"
        style={
          Object {
            "WebkitTransform": "translateY(100px)",
            "transform": "translateY(100px)",
          }
        }
      >
        <h6
          className="c7"
        >
          Go for a walk
        </h6>
        <p
          className="c8"
        >
          repeats every 
          8 hour
        </p>
        <div
          className="c9"
        >
          <button
            className="c10"
            onClick={[Function]}
            title="Start this nag"
          >
            <i
              className="c11"
            />
          </button>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<Nags /> populated nag list should match snapshot 1`] = `
.c17 {
  border: 1px solid undefined;
  cursor: pointer;
  display: inline-block;
  padding: 0;
  box-sizing: border-box;
  font-size: 0.85rem;
  line-height: 2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 0.5s;
  transition-duration: 0.5s;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  border-radius: 50%;
  height: 2rem;
  text-align: center;
  width: 2rem;
}

.c17 i {
  position: relative;
  top: 2px;
}

.c18 {
  font-family: icons;
  font-size: 1rem;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  line-height: 1;
  speak: none;
  text-transform: none;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c18:before {
  content: "\\e906";
}

.c20 {
  font-family: icons;
  font-size: 1rem;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  line-height: 1;
  speak: none;
  text-transform: none;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c20:before {
  content: "\\e9ac";
}

.c22 {
  font-family: icons;
  font-size: 1rem;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  line-height: 1;
  speak: none;
  text-transform: none;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c22:before {
  content: "\\ea1d";
}

.c25 {
  font-family: icons;
  font-size: 1rem;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  line-height: 1;
  speak: none;
  text-transform: none;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c25:before {
  content: "\\ea1c";
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 5px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 {
  position: relative;
}

.c2 input {
  padding-left: 20px;
}

.c2:before {
  content: '\\e986';
  font-family: icons;
  font-size: 0.5rem;
  left: 7px;
  position: absolute;
  top: 7px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  border: 1px solid;
  border-radius: 5px;
  box-sizing: border-box;
  font-size: .85rem;
  height: 1.5rem;
  padding: 0 0.35rem;
  width: 100%;
}

.c5 {
  margin-left: 5px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c7 {
  font-size: .8rem;
  line-height: 1.5rem;
  margin-right: 5px;
}

.c8 {
  border: 1px solid;
  box-sizing: border-box;
  font-size: .85rem;
  height: 1.5rem;
  margin-top: 0;
  margin-bottom: 0;
}

.c9 {
  list-style-type: none;
  margin: 0;
  overflow: hidden;
  padding: 0.5rem;
  will-change: transform;
}

.c10 {
  cursor: pointer;
  margin-bottom: 0.5rem;
  position: relative;
}

.c10:last-child {
  margin-bottom: 0;
}

.c11 {
  background-color: #ffffff;
  border-radius: 4px;
  box-shadow: 0 0 2px rgba(0,0,0,0.25);
  padding: 10px;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  will-change: transform;
  border-bottom: 3px solid undefined;
}

.c23 {
  border-radius: 4px;
  box-shadow: 0 0 2px rgba(0,0,0,0.25);
  padding: 10px;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  will-change: transform;
}

.c12 {
  font-size: 1.2rem;
  margin: 0;
}

.c13 {
  font-size: 0.85rem;
  margin-bottom: 0;
  margin-top: 5px;
}

.c14 {
  font-size: 0.75rem;
  margin-bottom: 0;
  margin-top: 5px;
}

.c15 {
  border-top: 3px solid #6e8ea1;
  border-radius: 0 0 3px 3px;
  bottom: -3px;
  left: 0;
  position: absolute;
  -webkit-transition: width 1s;
  transition: width 1s;
  will-change: width;
}

.c16 {
  position: absolute;
  right: -45px;
  top: 50%;
  will-change: transform;
}

.c19 {
  position: absolute;
  right: -85px;
  top: 50%;
  will-change: transform;
}

.c21 {
  position: absolute;
  right: -125px;
  top: 50%;
  will-change: transform;
}

.c24 {
  font-size: 0.85rem;
  font-weight: bold;
  margin-bottom: 0;
  margin-top: 5px;
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <div
          className="c3"
          name="query"
          onChange={[Function]}
          placeholder="Search"
          type="search"
          value=""
        >
          <input
            className="c4"
            id={undefined}
            name="query"
            onChange={[Function]}
            placeholder="Search"
            type="search"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="c5"
    >
      <div
        className="c6"
        name="sortBy"
        onChange={[Function]}
        value=""
      >
        <label
          className="c7"
          htmlFor="nags_sort"
          name="sortBy"
          onChange={[Function]}
          value=""
        >
          Sort By
        </label>
        <select
          className="c8"
          id="nags_sort"
          name="sortBy"
          onChange={[Function]}
          value=""
        >
          <option
            value="nextNag"
          >
            Next Nag
          </option>
          <option
            value="lastNag"
          >
            Last Nag
          </option>
          <option
            value="AZ"
          >
            A-Z
          </option>
          <option
            value="ZA"
          >
            Z-A
          </option>
          <option
            value="latest"
          >
            Latest First
          </option>
          <option
            value="oldest"
          >
            Oldest First
          </option>
        </select>
      </div>
    </div>
  </div>
  <ul
    className="c9"
  >
    <li
      className="c10"
      style={
        Object {
          "WebkitTransform": "translateY(100px)",
          "transform": "translateY(100px)",
        }
      }
    >
      <div
        onClick={[Function]}
      >
        <div>
          <div
            className="c11"
            style={
              Object {
                "WebkitTransform": "translateX(0px)",
                "transform": "translateX(0px)",
              }
            }
          >
            <h3
              className="c12"
            >
              Test Nag 1
            </h3>
            <div>
              <p
                className="c13"
              >
                in 
                48 years
              </p>
              <p
                className="c14"
              >
                <span>
                  and repeat every 
                  1 minute
                </span>
              </p>
            </div>
            <div
              className="c15"
              style={
                Object {
                  "width": "2526655556.85%",
                }
              }
            />
          </div>
        </div>
        <div>
          <div
            className="c16"
            style={
              Object {
                "WebkitTransform": "translate(0px, -50%)",
                "transform": "translate(0px, -50%)",
              }
            }
          >
            <button
              className="c17"
              onClick={[Function]}
              title="Edit"
            >
              <i
                className="c18"
              />
            </button>
          </div>
        </div>
        <div>
          <div
            className="c19"
            style={
              Object {
                "WebkitTransform": "translate(0px, -50%)",
                "transform": "translate(0px, -50%)",
              }
            }
          >
            <button
              className="c17"
              onClick={[Function]}
              title="Delete"
            >
              <i
                className="c20"
              />
            </button>
          </div>
        </div>
        <div>
          <div
            className="c21"
            style={
              Object {
                "WebkitTransform": "translate(0px, -50%)",
                "transform": "translate(0px, -50%)",
              }
            }
          >
            <button
              className="c17"
              onClick={[Function]}
              title="Pause"
            >
              <i
                className="c22"
              />
            </button>
          </div>
        </div>
      </div>
    </li>
    <li
      className="c10"
      style={
        Object {
          "WebkitTransform": "translateY(100px)",
          "transform": "translateY(100px)",
        }
      }
    >
      <div
        onClick={[Function]}
      >
        <div>
          <div
            className="c23"
            style={
              Object {
                "WebkitTransform": "translateX(0px)",
                "transform": "translateX(0px)",
              }
            }
          >
            <h3
              className="c12"
            >
              Test Nag 2
            </h3>
            <p
              className="c24"
            >
              Paused
            </p>
          </div>
        </div>
        <div>
          <div
            className="c16"
            style={
              Object {
                "WebkitTransform": "translate(0px, -50%)",
                "transform": "translate(0px, -50%)",
              }
            }
          >
            <button
              className="c17"
              onClick={[Function]}
              title="Edit"
            >
              <i
                className="c18"
              />
            </button>
          </div>
        </div>
        <div>
          <div
            className="c19"
            style={
              Object {
                "WebkitTransform": "translate(0px, -50%)",
                "transform": "translate(0px, -50%)",
              }
            }
          >
            <button
              className="c17"
              onClick={[Function]}
              title="Delete"
            >
              <i
                className="c20"
              />
            </button>
          </div>
        </div>
        <div>
          <div
            className="c21"
            style={
              Object {
                "WebkitTransform": "translate(0px, -50%)",
                "transform": "translate(0px, -50%)",
              }
            }
          >
            <button
              className="c17"
              onClick={[Function]}
              title="Resume"
            >
              <i
                className="c25"
              />
            </button>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
